### PR TITLE
test: integrationstests for ChargeLinksController

### PR DESCRIPTION
**Pull request by lasrinnil**
## Description

This PR will add integration tests for the ChargeLinksController.

## References

* #827

### CODE REVIEW

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/IntegrationTests/WebApi/ChargeLinksControllerTests.cs [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+            // Assert
+            var contentType = response.Content.Headers.ContentType!.ToString();
+            response.StatusCode.Should().Be(HttpStatusCode.OK);
+            contentType.Should().Be("application/json; charset=utf-8");
+        }
+
+        [Theory]
+        [InlineData("571313180000000005")]
+        public async Task GetAsync_WhenMeteringPointIdHasChargeLinks_ReturnsOrderedChargeLinks(string meteringPointId)
```
Comment (BjarkeMeier) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/IntegrationTests/WebApi/ChargeLinksControllerTests.cs:57]: How does this method test the ordering?

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/IntegrationTests/WebApi/ChargeLinksControllerTests.cs [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+        {
+            // Arrange
+            var client = _factory.CreateClient();
+            var expectedChargeLinks = 5;
```
Comment (BjarkeMeier) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/IntegrationTests/WebApi/ChargeLinksControllerTests.cs:61]: Is this vulnerable to changing test data? How about just ensuring that there is:
- at least (e.g.) 3 charge links
- all the charge links (regardless of how many there are) are ordered

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/IntegrationTests/WebApi/ChargeLinksControllerTests.cs [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+            // Act
+            var response = await client.GetAsync($"{BaseUrl}{meteringPointId}");
+            var jsonString = await response.Content.ReadAsStringAsync();
+            var actual = JsonSerializer.Deserialize<List<ChargeLinkDto>>(jsonString, new JsonSerializerOptions { PropertyNameCaseInsensitive = true });
+
+            // Assert
+            actual!.Count.Should().Be(expectedChargeLinks);
+            actual![3].ChargeId.Should().Be(expectedChargeId);
+            actual![3].StartDate.Should().Be(expectedStart);
```
Comment (BjarkeMeier) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/IntegrationTests/WebApi/ChargeLinksControllerTests.cs:73]: What is the purpose of these two assert considering the method name?

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/IntegrationTests/WebApi/ChargeLinksControllerTests.cs [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+        }
+
+        [Theory]
+        [InlineData("404")]
```
Comment (BjarkeMeier) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/IntegrationTests/WebApi/ChargeLinksControllerTests.cs:77]: Why not use `Fact` (applies to all the tests)?
Also in this case I'd prefer e.g. variable name `nonExistingMeteringPointId` rather that just `meteringPointId` as it is something I currently have to more or less guess. And btw, how about using `Guid.NewGuid()`, then you'd be almost certain that the metering point doesn't exist. And you could see it directly from the code without knowing whether "404" or not.

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/IntegrationTests/WebApi/ChargeLinksControllerTests.cs [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+        [Theory]
+        [InlineData("")]
+        public async Task GetAsync_WhenNoMeteringPointIdInput_ReturnsBadRequest(string meteringPointId)
+        {
+            var client = _factory.CreateClient();
+
+            var response = await client.GetAsync($"{BaseUrl}{meteringPointId}");
+
+            response.StatusCode.Should().Be(HttpStatusCode.BadRequest);
+        }
```
Comment (BjarkeMeier) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/IntegrationTests/WebApi/ChargeLinksControllerTests.cs:96]:
```suggestion
        [Fact]
        public async Task GetAsync_WhenNoMeteringPointIdInput_ReturnsBadRequest()
        {
            var missingMeteringPointId = string.Empty;
            var client = _factory.CreateClient();

            var response = await client.GetAsync($"{BaseUrl}{missingMeteringPointId}");

            response.StatusCode.Should().Be(HttpStatusCode.BadRequest);
        }
```

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/IntegrationTests/WebApi/ChargeLinksControllerTests.cs [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+using Xunit;
+using Xunit.Categories;
+
+namespace GreenEnergyHub.Charges.IntegrationTests.IntegrationTests.WebApi
+{
+    [IntegrationTest]
+    public class ChargeLinksControllerTests : WebApiHost, IClassFixture<ChargesDatabaseFixture>
+    {
+        private const string BaseUrl = "/ChargeLinks/GetAsync?meteringPointId=";
+        private const string KnownMeteringPointId = "571313180000000005";
```
Comment (BjarkeMeier) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/IntegrationTests/WebApi/ChargeLinksControllerTests.cs:35]: Known MP is a field while the missing is local...

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/IntegrationTests/WebApi/ChargeLinksControllerTests.cs [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+            // Assert
+            var contentType = response.Content.Headers.ContentType!.ToString();
+            response.StatusCode.Should().Be(HttpStatusCode.OK);
+            contentType.Should().Be("application/json; charset=utf-8");
+        }
+
+        [Fact]
+        public async Task GetAsync_WhenMeteringPointIdHasChargeLinks_ReturnsOrderedChargeLinks()
+        {
+            // Act
```
Comment (BjarkeMeier) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/IntegrationTests/WebApi/ChargeLinksControllerTests.cs:59]: AAA comments needs fix

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/IntegrationTests/WebApi/ChargeLinksControllerTests.cs [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+            var response = await _client.GetAsync($"{BaseUrl}{KnownMeteringPointId}");
+            var jsonString = await response.Content.ReadAsStringAsync();
+            var actual = JsonSerializer.Deserialize<List<ChargeLinkDto>>(
+                jsonString,
+                new JsonSerializerOptions
+                {
+                    PropertyNameCaseInsensitive = true,
+                    Converters = { new JsonStringEnumConverter() },
+                });
```
Comment (BjarkeMeier) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/IntegrationTests/WebApi/ChargeLinksControllerTests.cs:68]: Seems like something we'd like to reuse in the future. How about starting by segregating the serializer configuration to a separate method - probably returning a configures serializer.

**Review by BjarkeMeier — APPROVED** [review]
LGTM except for a few minor comments.

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/IntegrationTests/WebApi/ChargeLinksControllerTests.cs (hunk 6 above)
Comment (lasrinnil) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/IntegrationTests/WebApi/ChargeLinksControllerTests.cs:35]: the missing is only used in one place, I think that makes sense

**Pull request merged**